app/views/unattended/provisioning_templates/snippet/preseed_netplan_generic_interface.erb

```diff
@@ -16,25 +16,33 @@ oses:
 <%- end -%>
         dhcp4: <%= @dhcp %>
         dhcp6: <%= @dhcp6 %>
-<%- if !@dhcp && !@subnet.nil? && !@interface.ip.nil? -%>
-        addresses: [ <%= @interface.ip %>/<%= @subnet.cidr %> ]
+<%-
+static_v4 = !@dhcp && !@subnet.nil? && !@interface.ip.nil?
+static_v6 = !@dhcp6 && !@subnet6.nil? && !@interface.ip6.nil?
+-%>
+<%- if static_v4 || static_v6 -%>
+        addresses:
+<%-   if static_v4 -%>
+          - <%= @interface.ip %>/<%= @subnet.cidr %>
+<%-   end -%>
+<%-   if static_v6 -%>
+          - <%= @interface.ip6 %>/<%= @subnet6.cidr %>
+<%-   end -%>
 <%-   if @subnet.gateway.present? -%>
```

Comment:
Previously `gateway4` was only shown if static v4 was used. I'm not sure a static gateway makes if you don't provide a static address.
```suggestion
<%-   if static_v4 && @subnet.gateway.present? -%>
```